workspace.spec

```diff
@@ -572,6 +572,378 @@ module Workspace {
+		CRediT contributor role taxonomy: https://credit.niso.org
+
+		string contributor_type - must be either 'person' or 'organization'
+		string contributor_id (optional) - persistent unique identifier for the contributor;
+			e.g. an ORCID for an individual, or a ROR ID for an organization.
+		string given_name (optional) - first name, for individuals whose names can be split
+			into given and surnames
+		string family_name (optional) - family name, for individuals whose names can be
+			split into given and surnames.
+		string name (optional) - contributor name, in cases where the contributor is an
+			organization or a person whose name cannot easily fit into the given/family
+			name structure.
+		list<Organization> affiliations (optional) - list of organizations with which the
+			contributor is affiliated
+		list<string> contributor_roles (optional) - list of roles played by the contributor
+			when working on the resource.
+			Valid contributor_role values:
+			"Datacite:ContactPerson"
+			"Datacite:DataCollector"
+			"Datacite:DataCurator"
+			"Datacite:DataManager"
+			"Datacite:Distributor"
+			"Datacite:Editor"
+			"Datacite:HostingInstitution"
+			"Datacite:Producer"
+			"Datacite:ProjectLeader"
+			"Datacite:ProjectManager"
+			"Datacite:ProjectMember"
+			"Datacite:RegistrationAgency"
+			"Datacite:RegistrationAuthority"
+			"Datacite:RelatedPerson"
+			"Datacite:Researcher"
+			"Datacite:ResearchGroup"
+			"Datacite:RightsHolder"
+			"Datacite:Sponsor"
+			"Datacite:Supervisor"
+			"Datacite:WorkPackageLeader"
+			"Datacite:Other"
+			"CRediT:conceptualization"
+			"CRediT:data-curation"
+			"CRediT:formal-analysis"
+			"CRediT:funding-acquisition"
+			"CRediT:investigation"
+			"CRediT:methodology"
+			"CRediT:project-administration"
+			"CRediT:resources"
+			"CRediT:software"
+			"CRediT:supervision"
+			"CRediT:validation"
+			"CRediT:visualization"
+			"CRediT:writing-original-draft"
+			"CRediT:writing-review-editing"
+	*/
+	typedef structure {
+		string contributor_type;
+		string name;
+		string given_name;
+		string family_name;
+		string contributor_id;
+		list<Organization> affiliations;
+		list<string> contributor_roles;
+	} Contributor;
+
+	/*
+		Represents the credit metadata associated with a workspace object.
+
+		Required fields are:
+		- identifier
+		- resource_type
+		- titles (one or more required)
+		- contributors (one or more required)
+		- versioning information: if the resource does not have an explicit version number, one or more dates should be supplied, ideally the date of resource publication and the last update (if applicable).
+
+		string identifier - persistent unique identifier for the resource.
+			Should be in the format
+			<database name>:<identifier within database>
+			e.g.	DOI:10.46936/10.25585/60000745
+				CrossRef:IsManifestationOf
+				GO:0005456
+				HGNC:7470
+		string license (optional) - usage license for the resource.
+		string resource_type - the type of resource being represented.
```

Comment:
Definitely need more explanation / examples here, I'd have no idea what was required here

Reply:
Could you define what expected for a `dataset` vs a `record`

Reply:
we could reference the datacite documentation, which contains a variety of resource types. https://support.datacite.org/docs/what-are-the-resource-types-for-datacite-dois (too bad they don't define them there). @ialarmedalien - why did you pick dataset and record?

Reply:
After further discussion with Elisha, this is going to be reverted to 'dataset', since that is the only use case we will be dealing with for the foreseeable future.

Reply:
in that case I have to ask if the field is necessary at all

Reply:
I would vote yes, as this might need to expand to other controlled vocabulary as the platform matures. e.g., KBase Narratives as preprints

Reply:
Can it be added later when it's needed, or can it default to dataset so everyone who uses the api doesn't have to fill in a field where there's no other choice

Reply:
I defer to @ialarmedalien and the work to version, but def can be set to default = dataset

Reply:
Yes, it can definitely be set to default to dataset. I would definitely want to keep the field in because it's an important part of the schema.

Reply:
Can we note the default in the spec?

Reply:
Done.